### PR TITLE
Syn docs with ibm_db2 stubs

### PR DESCRIPTION
**Pull request by Girgias**
Supersedes #2300.

Generated using: ``php ../../php-src/build/gen_stub.php --replace-methodsynopses ../../ibm_db2/ibm_db2.stub.php ./reference/ibm_db2/``

### CODE REVIEW

**Comment by Girgias** [comment]
@NattyNarwhal could you review this as you seems to be the maintainer of the extension currently.

It would probably make a lot of sense to add various XML entities in language-snippets to prevent the needless repetitions for standard parameters across functions.

File: reference/ibm_db2/functions/db2-stmt-error.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   &reftitle.description;
   <methodsynopsis>
    <type>string</type><methodname>db2_stmt_error</methodname>
-   <methodparam choice="opt"><type>resource</type><parameter>stmt</parameter></methodparam>
+   <methodparam choice="opt"><type>resource</type><parameter>stmt</parameter><initializer>&null;</initializer></methodparam>
```
[the remark this answers is not in the document]
Comment (kocsismate) [reply reference/ibm_db2/functions/db2-stmt-error.xml:15]: @Girgias IIRC implicit nullability is not supported. I.e. you should get `Parameter $varName has null default, but is not nullable` exceptions.

[the remark this answers is not in the document]
Comment (Girgias) [reply reference/ibm_db2/functions/db2-stmt-error.xml:15]: Well I didn't really get any issues from generating the method synopsis via gen_stub. But I suppose this is something that needs patching upstream then.

**Comment by NattyNarwhal** [comment]
Discussion upstream about the autocommit stuff: https://github.com/php/pecl-database-ibm_db2/pull/32

**Comment by NattyNarwhal** [comment]
Discussion in the linked issue brought up gen-stub compatibility - will generated headers work on older PHP 8.x, even if it has newer constructs (for documentation and such) that appear in newer PHP? I know about the legacy arginfo header for 7.x, but I don't know what guarantees are present for 8.x. That'd affect what constructs we can put into the stub, and thus seemingly the documentation?

File: reference/ibm_db2/functions/db2-stmt-error.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (phansys) [reply reference/ibm_db2/functions/db2-stmt-error.xml:15]: @Girgias, do you have a suggestion about how to represent optional parameters in a stub that don't have a default value in the underlying implementation?
Currently, the stub is using `null` as default value for them (even when it is not true). Maybe there is a better way to represent that situation.

Thank you!

[the remark this answers is not in the document]
Comment (phansys) [reply reference/ibm_db2/functions/db2-stmt-error.xml:15]: BTW, I proposed [this](https://github.com/php/pecl-database-ibm_db2/pull/32#discussion_r1113680496) as a possible "convention".

[the remark this answers is not in the document]
Comment (kocsismate) [reply reference/ibm_db2/functions/db2-stmt-error.xml:15]: I answered here: https://github.com/php/pecl-database-ibm_db2/pull/32#discussion_r1114550575

**Comment by phansys** [comment]
@Girgias, the stub [`ibm_db2.stub.php`](https://github.com/php/pecl-database-ibm_db2/blob/master/ibm_db2.stub.php) was updated.

File: reference/ibm_db2/functions/db2-bind-param.xml [hunk 2]
```diff
@@ -12,10 +11,10 @@
   &reftitle.description;
   <methodsynopsis>
    <type>bool</type><methodname>db2_bind_param</methodname>
-   <methodparam><type>resource</type><parameter>stmt</parameter></methodparam>
+   <methodparam><type>resource</type><parameter>connection</parameter></methodparam>
```
Comment (phansys) [inline reference/ibm_db2/functions/db2-bind-param.xml:14]: I'm sorry. This parameter name is wrong.

php/pecl-database-ibm_db2#39.

Comment (phansys) [reply reference/ibm_db2/functions/db2-bind-param.xml:14]: The related PR was merged.

File: reference/ibm_db2/functions/db2-statistics.xml [hunk 3]
```diff
@@ -13,10 +13,10 @@
...
    <type>resource</type><methodname>db2_statistics</methodname>
    <methodparam><type>resource</type><parameter>connection</parameter></methodparam>
-   <methodparam><type>string</type><parameter>qualifier</parameter></methodparam>
-   <methodparam><type>string</type><parameter>schema</parameter></methodparam>
-   <methodparam><type>string</type><parameter>table-name</parameter></methodparam>
-   <methodparam><type>bool</type><parameter>unique</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>null</type></type><parameter>qualifier</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>null</type></type><parameter>schema</parameter></methodparam>
+   <methodparam><type>string</type><parameter>table_name</parameter></methodparam>
+   <methodparam><type class="union"><type>bool</type><type>int</type></type><parameter>unique</parameter></methodparam>
```
Comment (phansys) [inline reference/ibm_db2/functions/db2-statistics.xml:19]:
```suggestion
   <methodparam><type>bool</type><parameter>unique</parameter></methodparam>
```
See php/pecl-database-ibm_db2#50.

File: reference/ibm_db2/functions/db2-statistics.xml [hunk 4]
```diff
@@ -13,9 +13,9 @@
   <methodsynopsis>
    <type>resource</type><methodname>db2_statistics</methodname>
    <methodparam><type>resource</type><parameter>connection</parameter></methodparam>
-   <methodparam><type>string</type><parameter>qualifier</parameter></methodparam>
-   <methodparam><type>string</type><parameter>schema</parameter></methodparam>
-   <methodparam><type>string</type><parameter>table-name</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>null</type></type><parameter>qualifier</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>null</type></type><parameter>schema</parameter></methodparam>
+   <methodparam><type>string</type><parameter>table_name</parameter></methodparam>
    <methodparam><type>bool</type><parameter>unique</parameter></methodparam>
```
Comment (phansys) [inline reference/ibm_db2/functions/db2-statistics.xml:19]: The description for this parameter should also be updated, as it currently mentions integers.

File: reference/ibm_db2/functions/db2-fetch-object.xml [hunk 5]
```diff
@@ -11,9 +11,9 @@
  <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis>
-   <type>object</type><methodname>db2_fetch_object</methodname>
+   <type class="union"><type>stdClass</type><type>false</type></type><methodname>db2_fetch_object</methodname>
    <methodparam><type>resource</type><parameter>stmt</parameter></methodparam>
-   <methodparam choice="opt"><type>int</type><parameter>row_number</parameter><initializer>-1</initializer></methodparam>
+   <methodparam choice="opt"><type class="union"><type>int</type><type>null</type></type><parameter>row_number</parameter><initializer>&null;</initializer></methodparam>
```
[the remark this answers is not in the document]
Comment (phansys) [reply reference/ibm_db2/functions/db2-fetch-object.xml:16]: Done (see php/pecl-database-ibm_db2#51).

File: reference/ibm_db2/functions/db2-statistics.xml (hunk 4 above)
Comment (Girgias) [reply reference/ibm_db2/functions/db2-statistics.xml:19]: Okay done, the weird values is because it's initially some C int constants.

**Comment by Girgias** [comment]
Hopefully this is now ready to be merged.

**Comment by Girgias** [comment]
ups forgot about this

**Pull request merged**